pytest_operator/plugin.py

```diff
@@ -150,8 +168,7 @@ def abort_on_fail(request):
         ops_test.aborted = True
 
 
-@pytest.fixture(scope="module")
-@pytest.mark.asyncio
+@pytest_asyncio.fixture(scope="module")
```

Comment:
what's this for?

Reply:
It looks like [`async_mode=auto` was added upstream](https://github.com/pytest-dev/pytest-asyncio#auto-mode) which removes the need for the `@pytest.mark.asyncio` as well as [our implementation of the same functionality](https://github.com/charmed-kubernetes/pytest-operator/blob/05c372b9ecca08a82e271d297b74bdc4907b775e/pytest_operator/plugin.py#L110-L116), which is awesome. Though I guess `legacy` is still the default, so we probably want to keep ours in place for now, though maybe we could just set the option via code and not have to implement that auto-marking ourselves.

Reply:
Still, might better be done as a separate PR, rather than a drive-by. (Unless it's needed here for some reason that I don't see?)

Reply:
No it's not needed here so I will remove this and also `async_mode=auto`.